modules/market/src/main/java/com/opengamma/strata/market/interpolator/SquareLinearCurveInterpolator.java

```diff
@@ -0,0 +1,199 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.market.interpolator;
+
+import java.io.Serializable;
+
+import com.opengamma.strata.collect.array.DoubleArray;
+
+/**
+ * The interpolation is linear on y^2. The interpolator is used for interpolation on variance for options.
+ * All values of y must be positive. 
+ */
+final class SquareLinearCurveInterpolator implements CurveInterpolator, Serializable {
+
+    /**
+     * The serialization version id.
+     */
+    private static final long serialVersionUID = 1L;
+    
+    /** Level below which the value is consider to be 0. 
+     * 
+     */
+    private static final double EPS = 1.0E-10;
+    
+    /**
+     * The interpolator name.
+     */
+    public static final String NAME = "SquareLinear";
+    /**
+     * The interpolator instance.
+     */
+    
+    public static final CurveInterpolator INSTANCE = new SquareLinearCurveInterpolator();
+
+    /**
+     * Restricted constructor.
+     */
+    private SquareLinearCurveInterpolator() {
+    }
+
+    // resolve instance
+    private Object readResolve() {
+      return INSTANCE;
+    }
+
+    //-------------------------------------------------------------------------
+    @Override
+    public String getName() {
+      return NAME;
+    }
+
+    @Override
+    public BoundCurveInterpolator bind(DoubleArray xValues, DoubleArray yValues) {
+      return new Bound(xValues, yValues);
+    }
+
+    //-----------------------------------------------------------------------
+    @Override
+    public String toString() {
+      return NAME;
+    }
+
+    //-------------------------------------------------------------------------
+    /**
+     * Bound interpolator.
+     */
+    static class Bound extends AbstractBoundCurveInterpolator {
+      private final double[] xValues;
+      private final double[] yValues;
+      private final int dataSize;
+      Bound(DoubleArray xValues, DoubleArray yValues) {
+        super(xValues, yValues);
+        this.xValues = xValues.toArrayUnsafe();
+        this.yValues = yValues.toArrayUnsafe();
+        dataSize = xValues.size();
+      }
+
+      Bound(Bound base, BoundCurveExtrapolator extrapolatorLeft, BoundCurveExtrapolator extrapolatorRight) {
+        super(base, extrapolatorLeft, extrapolatorRight);
+        this.xValues = base.xValues;
+        this.yValues = base.yValues;
+        dataSize = xValues.length;
+      }
+
+      //-------------------------------------------------------------------------
+      @Override
+      double doInterpolate(double xValue) {
+        int lowerIndex = lowerBoundIndex(xValue, xValues);
```

Comment:
it is worth keeping the comments from `LinearCurveInterpolator`.

`doInterpolate` has

          // x-value is less than the x-value of the last node (lowerIndex < intervalCount)

which is a reminder that xValue is strictly less than xValues[xValues.length - 1]

whereas the comments in `doFirstDerivative` and `doParameterSensitivity` remind that xValue might equal xValues[xValues.length - 1]